### PR TITLE
PagerDuty-Demo

### PR DESCRIPTION
**Pull request by matthewzhaocc**

### CODE REVIEW

File: examples/pagerduty/index.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    finishedUrl: '/finished',
+    mapToUserId: (req) => req.params.userId,
+  })
+);
+
+app.get('/finished', handleSession, async (req, res) => {
+  const creds = await everyauth.getIdentity('pagerduty', req.session.userId);
+  const sdk = api({ token: creds.accessToken, tokenType: 'bearer' });
+  const services = await (await sdk.get('/services')).resource;
+  const me = await (await sdk.get('/users/me')).data.user;
```
Comment (degrammer) [inline examples/pagerduty/index.js:53]: Seems there is an extra await here?

Comment (degrammer) [reply examples/pagerduty/index.js:53]: Also, I'm wondering if a new incident can be created from the API? would be an interesting part of this demo!

File: examples/pagerduty/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "serve": "nodemon index.js"
+  },
+  "keywords": [],
+  "author": "Fusebit, Inc.",
+  "license": "ISC",
+  "dependencies": {
+    "@fusebit/everyauth-express": "^1.0.3",
```
Comment (degrammer) [inline examples/pagerduty/package.json:15]: Change to "@fusebit/everyauth-express": "*"

**Review by degrammer — CHANGES_REQUESTED** [review]
Review comments, also delete package.lock file, we don't include them in examples.

File: examples/pagerduty/README.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+## Handler
+
+Next, you need to build a handler to be call back to after the authorization is processed, in the above code, we defined it to be `/finished`. So here, we want to add a `/finished` route to get the PagerDuty credentials and use it to call different APIs, add this to your code:
+
+```javascript
+app.get('/finished', handleSession, async (req, res) => {
+  const creds = await everyauth.getIdentity('pagerduty', req.session.userId);
+  const sdk = api({ token: creds.accessToken, tokenType: 'bearer' });
+  const services = await (await sdk.get('/services')).resource;
+  const me = await (await sdk.get('/users/me')).data.user;
```
Comment (degrammer) [inline examples/pagerduty/README.md:74]: Organize this code in blocks/sections, so is easier to follow what each line is doing, e.g:
1. Get the stored credentials for the pagerduty service
2. Fetch the authorizing user profile, etc

See an example https://github.com/degrammer/everyauth-express/blob/everyauth-stackoverflow/examples/stackoverflow/index.js#L67

**Review by matthewzhaocc — COMMENTED** [review]
This is the coolest ever!

File: examples/pagerduty/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+![Screenshot demo](pd-demo.png "Screenshot demo")
+## Install dependencies
+
+```bash
+npm i
+```
+
+Run the application
+
+```bash\
```
Comment (degrammer) [inline examples/pagerduty/README.md:23]: remove extra \

**Review by degrammer — CHANGES_REQUESTED** [review]
Small comment and also, can we remove any package-lock from the examples, I'm seeing some of them are being added

File: examples/pagerduty/README.md (hunk 4 above)
Comment (matthewzhaocc) [reply examples/pagerduty/README.md]: Done

**Review by degrammer — APPROVED** [review]
LGTM

**Pull request merged**